classes/local/aws_helper.php

```diff
@@ -61,4 +65,36 @@ public static function get_proxy_string() {
         }
         return $proxy;
     }
+
+    /**
+     * Configure the provided AWS client to route traffic via the moodle proxy for any hosts not excluded.
+     *
+     * @param AwsClient $client
+     * @return AwsClient
+     */
+    public static function configure_client_proxy(AwsClient $client): AwsClient {
+        $client->getHandlerList()->appendBuild(self::get_callable_middleware(), 'proxy_bypass');
+        return $client;
+    }
+
+    /**
+     * Generate a middleware higher order function to wrap the handler and append proxy configuration based on target.
+     *
```

Comment:
Might want to add function parameter phpdoc here for completeness.

Reply:
This function has no params. The callable return takes a parameter, but not the generating function.